src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldType.java

```diff
@@ -74,11 +91,134 @@ public Query termQuery(Object value, QueryShardContext context) {
+        return cachedKNNVectorFieldTypeConfig.getKnnVectorFieldTypeConfig().getDimension();
+    }
+
+    /**
+     * Get the vector data type of the field
+     *
+     * @return the vector data type of the field
+     */
+    public VectorDataType getVectorDataType() {
+        return cachedKNNVectorFieldTypeConfig.getKnnVectorFieldTypeConfig().getVectorDataType();
+    }
+
+    /**
+     * Get the model id if the field is configured to have it. Null otherwise.
+     *
+     * @return the model id if the field is built for ann-indexing, empty otherwise
+     */
+    public Optional<String> getModelId() {
+        return Optional.ofNullable(modelId);
+    }
+
+    /**
+     * Determine whether the field is built for ann-indexing. If not, only brute force search is available
+     *
+     * @return true if the field is built for ann-indexing, false otherwise
+     */
+    public boolean isIndexedForAnn() {
+        return getModelId().isPresent() || getKNNIndexContext().isPresent();
+    }
+
+    /**
+     * Return a map of query parameters that are valid for the given query context and augmented with other
+     * parameters
+     *
+     * @param queryContext Context of the query
+     * @param originalMethodParameters user provided query parameters
+     * @return parameters to be passed to the library augmented based on the field type
+     */
+    public Map<String, Object> getProcessedQueryMethodParameters(QueryContext queryContext, Map<String, Object> originalMethodParameters) {
+        if (originalMethodParameters == null || originalMethodParameters.isEmpty()) {
+            return originalMethodParameters;
+        }
+
+        // If we are unable to get the configuration and the user is trying to passs in parameters, we have to fail
+        // the request
+        KNNIndexContext knnIndexContext = getKNNIndexContext().orElseThrow(
+            () -> new IllegalArgumentException(
+                "Unable to validate passed in method parameters because index was built with model before 2.14"
+            )
+        );
+
+        final KNNLibrarySearchContext engineSpecificMethodContext = knnIndexContext.getKnnLibrarySearchContext();
+        return engineSpecificMethodContext.processMethodParameters(queryContext, originalMethodParameters);
+    }
+
+    public RescoreContext getProcessedRescoreQueryContext(QueryContext queryContext, RescoreContext originalRescoreContext) {
+        if (originalRescoreContext != null) {
+            return originalRescoreContext;
+        }
+        Optional<KNNIndexContext> knnIndexContext = getKNNIndexContext();
+        return knnIndexContext.map(indexContext -> indexContext.getKnnLibrarySearchContext().getDefaultRescoreContext(queryContext))
+            .orElse(RescoreContext.DISABLED_RESCORE_CONTEXT);
+    }
+
+    Optional<KNNIndexContext> getKNNIndexContext() {
+        KNNVectorFieldTypeConfig knnVectorFieldTypeConfig = cachedKNNVectorFieldTypeConfig.getKnnVectorFieldTypeConfig();
```

Comment:
we really don't have to do whole if else , if we can make optional returnbable from cachedKNNVectorFieldTypeConfig